test/cql-pytest/test_group_by_limit.py

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import asyncio
+import time
+import pytest
+import logging
+from cassandra.protocol import InvalidRequest, ReadFailure                         # type: ignore
+from cassandra.query import SimpleStatement                                        # type: ignore
+
+
+logger = logging.getLogger(__name__)
+
+@pytest.mark.asyncio
+async def test_add_index(cql):
+    assert cql is not None
+    cql.execute("create keyspace ks with replication = {'class': 'SimpleStrategy', 'replication_factor': 1};")
+    cql.set_keyspace("ks")
+    cql.execute("""create table grouping (
+        a int,
+        b int,
+        c int,
+        PRIMARY KEY (a, b)
+        ) WITH CLUSTERING ORDER BY (b ASC);""")
+
+    pk = 0
+    primes = [13, 17, 19, 23]
+
+    for i in range(29):
+        cql.execute(f"insert into grouping (a,b,c) values ({29},{i+1},{primes[i%4]});");
+
+    for i in range(31):
+        cql.execute(f"insert into grouping (a,b,c) values ({31},{i+1},{primes[i%4]});");
+
+    for i in range(37):
+        cql.execute(f"insert into grouping (a,b,c) values ({37},{i+1},{primes[i%4]});");
+
+    for i in range(41):
+        cql.execute(f"insert into grouping (a,b,c) values ({41},{i+1},{primes[i%4]});");
+
+    ret = set(cql.execute("select * from grouping group by a"))
+
+    assert len(ret) == 4
+    assert set([29, 31, 37, 41]) == set([x[0] for x in ret])
+
+    ret = set(cql.execute("select * from grouping group by a limit 1"))
+
+    assert len(ret) == 1    
+    assert ret.pop()[0] in [29, 31, 37, 41]
+
+    ret = set(cql.execute("select COUNT(*) from grouping group by a"))
+
+    assert len(ret) == 4
+    assert set([29, 31, 37, 41]) == set([x[0] for x in ret])
+
+    ret = set(cql.execute("select COUNT(*) from grouping group by a limit 1"))
```

Comment:
how about trying `limit` with other value than `1`?

Reply:
Maybe in a loop starting from 1 to N? Where N = all records. Is `limit 0` a valid option?